### DISCUSSION TITLE
Should `.meta` be providing more than only `.license`?

### DISCUSSION THREAD

**Discussion by mightyiam** [90-008/nix-cargo-integration#181] [Q&A] [answered]
Hey, are executable packages expected to end up with anything other than `.license` under `.meta`?

Here's a reproduction from our work-in-progress migration of a project from Crane to NCI:

```
$ nix eval github:molybdenumsoftware/pr-tracker/ad3ea580a3cfd6d5b1f2ae9f58f600b5d09a027d#packages.x86_64-linux.api.meta
{ license = [ { deprecated = false; free = true; fullName = "GNU Affero General Public License v3.0 only"; redistributable = true; shortName = "agpl3Only"; spdxId = "AGPL-3.0-only"; url = "https://spdx.org/licenses/AGPL-3.0-only.html"; } ]; }
```

**Comment by 90-008** [comment]
it also puts description and homepage if the package / workspace has it. there isn't any other info we can extract from a Cargo.toml that would be useful i believe.

**Reply by mightyiam** [reply]
Crane provides `.mainProgram`. Can we have that?

**Reply by 90-008** [reply]
it does? dream2nix already uses crane, but i don't think it exposes any such info? i don't see anything with mainProgram in crane source either. regardless i think it's pretty easy to just add that info yourself either in `mkDerivation.meta` or otherwise

**Reply by mightyiam** [reply]
Sorry, I take that back. In the following project, where Crane is used directly, `mainProgram` is explicitly set. But there are some more attrs in `.meta`:
```
$ nix eval github:molybdenumsoftware/pr-tracker/e686279b6e4665c370e62a42b6813f80962b6884#api.meta
{ available = true; broken = false; insecure = false; mainProgram = "pr-tracker-api"; name = "pr-tracker-api-4.1.1"; outputsToInstall = [ "out" ]; position = "/nix/store/spdyfwr32dirq54x0k29ysr00x1n8gjb-source/lib/mkCargoDerivation.nix:62"; unfree = false; unsupported = false; }
```

**Reply by 90-008** [reply, marked as answer]
those aren't really things we can glean from the Cargo.toml or cargo metadata itself, or only really matter for nixpkgs, don't think nci will / should try to set any of those. and any of these can just be set by the user very simply so there isn't any need for nci to do anything here.
